lib/services/docker.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Package for cups service tests
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+package services::docker;
+use base 'opensusebasetest';
+use testapi;
+use utils;
+use strict;
+use warnings;
+use containers::docker;
+use containers::utils 'basic_container_tests';
+use containers::container_images 'build_and_run_image';
+
+my $service_type = 'Systemd';
+
+# check service is running and enabled
+sub check_service {
+    common_service_action 'docker', $service_type, 'is-enabled';
+    common_service_action 'docker', $service_type, 'is-active';
+}
+
+# check docker service before and after migration
+# stage is 'before' or 'after' system migration.
+sub full_docker_check {
+    my (%hash) = @_;
+    my ($stage, $type) = ($hash{stage}, $hash{service_type});
```

Comment:
The $stage hasn't been used, means do the same test steps before and after migration, right?

Reply:
The `$stage` is used in line 35:
```perl
$engine->init() if ($stage eq 'before');
```

Reply:
Ok, so the only different is before migration need init the engine. Thanks.

Reply:
Yes, we need `$engine` in both stages but we install and prepare docker only once.